Remove assessment search empty query param validation

aliuk2012 opened the pull request:
# Context

This was removed because it introduced a new "validation" pattern in the service and didn't match what GOV.UK design system recommends. On the off chance the user removes a previous search terms and submits the form they would get this copy telling them that they cant search without text when it could just be seen as "clearing" the search filter

aliuk2012 committed:
Remove assessment search empty query param validation

This was removed because it introduced a new "validation" pattern in the service and didn't match what GOV.UK design system recommends. On the off chance the user removes a previous search terms and submits the form they would get this copy telling them that they cant search without text when it could just be seen as "clearing" the search filter

froddd commented on server/controllers/temporary-accommodation/manage/bookingSearchController.test.ts:
```diff
@@ -162,38 +158,11 @@ describe('BookingSearchController', () => {
-        'renders an error for %s bookings',
-        async uiStatus => {
-          const status = (uiStatus === 'active' ? 'arrived' : uiStatus) as BookingSearchApiStatus
-          const searchParameters = bookingSearchParametersFactory.build({ crn: '   ' })
```
The previous implementation trimmed the CRN to assert it had a minimum length -- this makes me wonder if we should actually do this for any input for this?

aliuk2012 replied:
I agree. I'm going to trim the users input as part of the rename the search param and completely ignore the param if its empty and treat it as if the user clicked on "Clear"

froddd commented on server/views/temporary-accommodation/assessments/index.njk:
```diff
@@ -14,7 +14,7 @@
 {# TODO Move this HTML to the searchByCrnOrNameResult once HTML can be standardised between referrals and bookings #}
 {% set resultsHtml %}
-    {% if (not context.errors.prn) and ((tableRows|length > 0) or (not crn)) %}
+    {% if tableRows|length > 0 %}
```
I believe we still need `or not crn` here, as otherwise we would show 'There are no results for ‘’ in ...' if there is nothing to show (e.g. no records for that view, at all). Possibly not the best UX in this case, but I think an empty table is less confusing than a message about an empty search the user haven't even submitted!

aliuk2012 replied:
I'll add it back for the moment while we improve the UX.

froddd commented on server/views/temporary-accommodation/booking-search/results.njk:
```diff
@@ -11,7 +11,7 @@
 {# TODO Move this HTML to the searchByCrnOrNameResult once HTML can be standardised between referrals and bookings #}
 {% set resultsHtml %}
-    {% if (not context.errors.prn) and ((response.data|length > 0) or (not crn)) %}
+    {% if response.data|length > 0 %}
```
Same as above, I think `or not crn` should still be here

aliuk2012 replied:
I'll add it back for the moment while we improve the UX.

froddd reviewed (changes requested):
Couple of comments for the case when there are genuinely no records but no search has been submitted.

aliuk2012 committed:
Reintroduce some logic into the resultsHTML

This is needed for the moment to show either any results or an empty table with only headings if the search term is empty

aliuk2012 committed:
Remove some test helper methods that were no longer used

froddd reviewed (approved):
👍
